src/api/db/gameSession.ts

```diff
@@ -20,3 +23,25 @@ export async function requestGetPlayerHand(
     A.reduce<Normalized<Card>, Normalized<Card>>({}, (acc, card) => ({ ...acc, ...card })),
   );
 }
+
+export async function requestPlayerHandListener(
+  currentPlayerId: string,
+  sessionId: string,
+  callback: (h: string[]) => void,
+) {
+  try {
+    await getSessionRef(sessionId)
+      .collection('players')
+      .doc(currentPlayerId)
+      .onSnapshot(documentSnapshot => {
+        const player = extractDocumentData<SessionPlayer>(documentSnapshot);
+
+        if (O.isSome(player)) {
```

Comment:
You should do a fold an throw an error if there is no player cause that is an impossible state.